test/pylib/report_plugin.py

```diff
@@ -26,6 +28,15 @@ def pytest_runtest_makereport(self):
         report = outcome.get_result()
         if self.mode is not None or self.run_id is not None:
             report.nodeid = f"{report.nodeid}.{self.mode}.{self.run_id}"
+        status =  get_pytest_report_status(report)
+        # skip attach logs for passed tests
+        if status != Status.PASSED and not self.config.option.attach_capture:
```

Comment:
Suggestion: perhaps we should consider checking the `--save-log-on-success`/`-s` flag and, if enabled, attaching the logs regardless of the result? Its current effect is that it leaves all logs on disk instead of leaving only those of failed tests. I think it will actually be surprising if the option does not have this effect, and it also would be nice to preserve logs of all tests if the developer wishes to.

Reply:
By the way, where is `self.config.option.attach_capture` documented? What is its meaning? I couldn't find information about it.

Reply:
Actually, it's a good idea. Now, allure will respect the `-s` parameter from the test.py.
`attach_capture` it's an option from the allure-plugin. Added comment about that.

Reply:
I see you copy-pasted the implementation from allure:
https://github.com/allure-framework/allure-python/blob/master/allure-pytest/src/listener.py#L235

that doesn't look like a good practice -- it depends on allure internals -- what if these internals change?

Is there no other way? I was thinking of reconfiguring allure (i.e. set attach_capture) and let allure do the job. However I guess that won't work, because the configuration is global for all test runs and all reports...

Reply:
The only thing that belongs to allure internal is method `get_pytest_report_status`. It's not fully copy-paste, I'm using there not internal way to attach logs, but public methods of [allure](https://allurereport.org/docs/pytest/#attach-screenshots-and-other-files). So, if the behavior changes, we will need to modify our code.
Regarding the allure capturing logs, you are right. It can attach all logs for all tests, or no logs at all. That's why we need this code.
Changed the providing type as offered in official documentation.